companies/views.py

```diff
@@ -429,18 +410,41 @@ def companies_list(request, year):
             }
         )
 
+    total_pages = filtered_companies.count() // COMPANIES_PER_PAGE
     return render(
         request,
         "companies/companies_list.html",
         {
             "fair": fair,
-            "companies": companies_modified,
-            "companies_ids": [x["pk"] for x in companies_modified],
+            "companies": Company_Page(companies_modified, total_pages, page_number),
```

Comment:
In Python we generally don't use capitalize the words in a snake case. this variable should preferably look like `company_page`

Reply:
Changed it to CompanyPage as it's a class name

Reply:
Great!